pinot-spi/src/main/java/org/apache/pinot/spi/data/FieldSpec.java

```diff
@@ -99,6 +100,7 @@ public abstract class FieldSpec implements Comparable<FieldSpec>, Serializable {
   protected String _name;
   protected DataType _dataType;
   protected boolean _isSingleValueField = true;
+  protected Boolean _nullable = null;
```

Comment:
Per our discussion, should we add it as primitive boolean `_notNull` and default to false?

Reply:
We have run into problems with default true configs before, so it is best practice to use false as default to prevent bugs because people usually assume the default is false